### ISSUE TITLE
Receiving empty output in captum insights eventhough insights flask server shows request success in log

### ISSUE THREAD

**Issue by KasirajanA** [pytorch/captum#686]
## ❓ Questions and Help

Hello,

Ran the [example.py](https://github.com/pytorch/captum/blob/bf8b15e1fb8bdc7b6e226d656b6a0c4d4825993c/captum/insights/example.py) after making below mentioned changes to formatted_data_iter function.

- Reduced batch size to 1
- Updated the function to only yield 4 results

```# Sample code
def formatted_data_iter():
    dataset = torchvision.datasets.CIFAR10(
        root="data/test", train=False, download=True, transform=transforms.ToTensor()
    )
    dataloader = iter(
        torch.utils.data.DataLoader(dataset, batch_size=1, shuffle=False, num_workers=1)
    )
    for i in range(4):
        images, labels = next(dataloader)
        yield Batch(inputs=images, labels=labels)
```

Only for first fetch request getting output in captum insights. From second fetch request getting empty page in captum insights eventhough getting http response with status code 200.

**Observation:**
For each fetch request in captum insights UI, requested number of samples is fetched from the data iter [(Ref in source)](https://github.com/pytorch/captum/blob/bf8b15e1fb8bdc7b6e226d656b6a0c4d4825993c/captum/insights/attr_vis/app.py#L442) Since the data iter is not reinitialised,  it is exploited when multiple fetch requests are made and it throws StopIteration exception. But the exception is caught and neglected [(Ref in source)](https://github.com/pytorch/captum/blob/bf8b15e1fb8bdc7b6e226d656b6a0c4d4825993c/captum/insights/attr_vis/app.py#L470). Because of which there is no data to process and we are getting empty response in captum insights. 

Is this a expected behaviour or a bug?

Attached the modified example.py for reference.
[example.zip](https://github.com/pytorch/captum/files/6682459/example.zip)

**Comment by NarineK** [comment]
cc: @Reubend

**Comment by Reubend** [comment]
Hi there @KasirajanA thanks for reporting this and my apologies for the slow reply. This is definitely a bug and I've just sent out #728 to address it.

**Comment by Reubend** [comment]
This should now be fixed by #728 
Let me know if there are any further issues.
